src/main/scala/com/databricks/spark/xml/XmlInputFormat.scala

```diff
@@ -253,7 +255,7 @@ private[xml] class XmlRecordReader extends RecordReader[LongWritable, Text] {
   private def checkAttributes(current: Int): Boolean = {
     var len = 0
     var b = current
-    while(len < space.length && b == space(len)) {
+    while (len < space.length && b == space(len) || newLineSpace.exists(_.apply(len) == b)) {
```

Comment:
> Really, I think this should just read while it can find a match for any whitespace character's bytes. (That would resolve the problem that this also doesn't allow multiple spaces before an attribute.) That means you can skip the \r\n case as it's contained in the \r and \n cases.

I check it on multiple space, and it's work correctly
```xml
<note

        
        xmlns="http://www.w3schools.com"

        
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://www.w3schools.com note.xsd">
    <to

    >

        Tove</to>
    <from>Jani</from>
    <heading>Reminder</heading>
    <body>Don't forget me this weekend!</body>
</note>
```

> And then that means we should probably peek at input here not consume it, which means a buffered input stream.

I don't see problem on this, because this library used for reading `xml` from `hdfs`.

>  Oh boy, I might take a crack at this myself as this might be a bigger issue!

Of course, this library may contains many hidden bugs.

Reply:
I'm just getting used to this code. I think the whitespace isn't an issue, as this part of the code is merely searching for the start and eng tags, and lets the STaX parser parse whatever is inside, so the whitespace is OK.

I also don't think the multi-byte issue is a problem because even if it reads too far, it means it has read into data it doesn't care about which will be skipped later anyway.

(Note this isn't just about HDFS, but the data store is irrelevant; the input might still have a particular different encoding.)

I think this still has an issue that you won't actually use your `\r\n` case. You just want to check the three possible next 'space' characters and if any matches, assume you've successfully found the open tag.